collector/dns.go

```diff
@@ -0,0 +1,492 @@
+// returns data points from Win32_PerfRawData_DNS_DNS
+// https://msdn.microsoft.com/en-us/library/ms803992.aspx?f=255&MSPPError=-2147217396
+// https://technet.microsoft.com/en-us/library/cc977686.aspx
+package collector
+
+import (
+	"log"
+
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	Factories["dns"] = NewDNSCollector
+}
+
+// A DNSCollector is a Prometheus collector for WMI Win32_PerfRawData_DNS_DNS metrics
+type DNSCollector struct {
+	ZoneTransferRequestsReceived  *prometheus.Desc
+	ZoneTransferRequestsSent      *prometheus.Desc
+	ZoneTransferResponsesReceived *prometheus.Desc
+	ZoneTransferSuccessReceived   *prometheus.Desc
+	ZoneTransferSuccessSent       *prometheus.Desc
+	ZoneTransferFailures          *prometheus.Desc
+	MemoryUsedBytes               *prometheus.Desc
+	DynamicUpdatesQueued          *prometheus.Desc
+	DynamicUpdatesReceived        *prometheus.Desc
+	DynamicUpdatesFailures        *prometheus.Desc
+	NotifyReceived                *prometheus.Desc
+	NotifySent                    *prometheus.Desc
+	SecureUpdateFailures          *prometheus.Desc
+	SecureUpdateReceived          *prometheus.Desc
+	Queries                       *prometheus.Desc
+	Responses                     *prometheus.Desc
+	RecursiveQueries              *prometheus.Desc
+	RecursiveQueryFailures        *prometheus.Desc
+	RecursiveQuerySendTimeouts    *prometheus.Desc
+	WinsQueries                   *prometheus.Desc
+	WinsResponses                 *prometheus.Desc
+	UnmatchedResponsesReceived    *prometheus.Desc
+}
+
+// NewDNSCollector ...
+func NewDNSCollector() (Collector, error) {
+	const subsystem = "dns"
+	return &DNSCollector{
+		ZoneTransferRequestsReceived: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "zone_transfer_requests_received_total"),
+			"Number of zone transfer requests (AXFR/IXFR) received by the master DNS server",
+			[]string{"type"},
```

Comment:
You'll have to forgive me, it's been about 15 years since I read the DNS RFCs. qtype would be better too.